mesonbuild/msetup.py

```diff
@@ -163,6 +164,8 @@ def add_vcs_ignore_files(self, build_dir: str) -> None:
 
     def validate_dirs(self, dir1: str, dir2: str, reconfigure: bool, wipe: bool) -> T.Tuple[str, str]:
         (src_dir, build_dir) = self.validate_core_dirs(dir1, dir2)
+        if Path(os.path.realpath(build_dir)) in Path(os.path.realpath(src_dir)).parents:
+            raise MesonException('Source directory must not be inside of the build directory.')
```

Comment:
os.path.realpath is not needed, validate_core_dirs() does it already.

Reply:
Done